File: 0724-find-pivot-index/0724-find-pivot-index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int n=nums.size();
        int sum=0;
        for(int i=0;i<n;i++){
            sum+=nums[i];
        }
        int lsum=0;
        int rsum=sum-nums[0];
        int j=1;
        int index=-1;
        while(j<n){
            if(lsum==rsum){
                index=j-1;
                break;
            }
            else{
                rsum-=nums[j];
                lsum+=nums[j-1];
                j++;
            }
        }
        if(lsum==rsum){
            index=j-1;
        }
        return index;

    }
};

File: 0724-find-pivot-index/0724-find-pivot-index.java
class Solution {
    public int pivotIndex(int[] nums) {
        int s=0;
        int n=nums.length;
        for(int i=0;i<n;i++){
            s+=nums[i];
        }
        int lsum=0;
        int rsum=s-nums[0];
        if(lsum==rsum) return 0;
        for(int i=1;i<n;i++){
            lsum+=nums[i-1];
            rsum-=nums[i];
            if(lsum==rsum) return i;
        }
        return -1;
    }
}